xbmc/interfaces/legacy/Player.cpp

```diff
@@ -42,12 +42,9 @@ namespace XBMCAddon
     {
       iPlayList = PLAYLIST_MUSIC;
 
-      if (_playerCore == EPC_VideoPlayer ||
-          _playerCore == EPC_MPLAYER ||
-          _playerCore == EPC_PAPLAYER)
-        playerCore = (EPLAYERCORES)_playerCore;
-      else
-        playerCore = EPC_NONE;
+      if (_playerCore != 0)
+        CLog::Log(LOGERROR, "xbmc.Player: Requested non-default player. This behavior is deprecated, plugins may no longer specify a player");
```

Comment:
Either you need to keep the constants in ModuleXbmc.h for backwards compatibility. or, if you are breaking the api, this and the constructor argument should be removed as well, as there is no way to pass the constants anyway.